### ISSUE TITLE
apigatewayv2-integrations: Add support for SQS-SendMessage integration

### ISSUE THREAD

**Issue by DaWyz** [aws/aws-cdk#24785]
### Describe the feature

ApiGatewayv2 HTTP API integrates with SQS to redirect the request into a queue. It would be nice to have CDK implement this. See [documentation](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-aws-services-reference.html#SQS-SendMessage) on `SQS-SendMessage` mapping.

### Use Case

I want an easy way to use an SQS Queue as a proxy for my HTTP API.

### Proposed Solution

Implement the construct similarly to the lambda integration.

It would look like the following:

```ts
    const stack = new Stack();
    const api = new HttpApi(stack, 'HttpApi');
    const queue = DefaultQueue(stack, 'Queue');
    new HttpRoute(stack, 'LambdaProxyRoute', {
      httpApi: api,
      integration: new HttpSqsQueueIntegration('Integration', queue, {
        parameterMapping: new ParameterMapping()
          .appendHeader('header2', MappingValue.requestHeader('header1'))
          .removeHeader('header1'),
      }),
      routeKey: HttpRouteKey.with('/pets'),
    });
```

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.70.0

### Environment details (OS name and version, etc.)

Ubuntu 20

**Comment by pahud** [comment]
Yes this would be awesome! We'd be glad to review your PR when it is ready. Thank you.

**Comment by benm5678** [comment]
Is there any workaround people are using in the meantime?  I'm trying to construct it via HttpIntegration, but having some issues and can't find working examples.

**Comment by benm5678** [comment]
In case it helps give a start, here's how I did it for the SendMessage:

Usage:

```
httpApi.addRoutes({
    path: `/sendMessage`,
    methods: [gw2.HttpMethod.POST],
    integration: new HttpSqsSendIntegration(this, 'MySQSSendIntegrationId', myQueue, {
      messageAttributes: '{ "test": { "DataType": "String", "StringValue": "val1" } }'
    }),
  });
```

Extended class:

```
import * as iam from 'aws-cdk-lib/aws-iam';
import * as gw2 from '@aws-cdk/aws-apigatewayv2-alpha';
import * as sqs from 'aws-cdk-lib/aws-sqs';
import { Construct } from 'constructs';

export interface HttpSqsSendIntegrationProps {
  readonly messageBody?: string;
  readonly messageAttributes?: string;
}

export class HttpSqsSendIntegration extends gw2.HttpRouteIntegration {
  private readonly sqs: sqs.Queue;
  private readonly role: iam.Role;
  private readonly props?: HttpSqsSendIntegrationProps;
  bind(options: gw2.HttpRouteIntegrationBindOptions): gw2.HttpRouteIntegrationConfig {
    return {
      type: gw2.HttpIntegrationType.AWS_PROXY,
      subtype: gw2.HttpIntegrationSubtype.SQS_SEND_MESSAGE,
      payloadFormatVersion: gw2.PayloadFormatVersion.VERSION_1_0,
      parameterMapping: new gw2.ParameterMapping()
        .custom('QueueUrl', this.sqs.queueUrl)
        .custom('MessageBody', this.props?.messageBody ?? '$request.body')
        .custom('MessageAttributes', this.props?.messageAttributes ?? ''),
      credentials: gw2.IntegrationCredentials.fromRole(this.role),
    };
  }

  constructor(scope: Construct, id: string, sqs: sqs.Queue, props?: HttpSqsSendIntegrationProps) {
    super(id);
    this.sqs = sqs;
    this.props = props;
    this.role = new iam.Role(scope, "HTTP-API-Role", {
      assumedBy: new iam.ServicePrincipal("apigateway.amazonaws.com"),
      inlinePolicies: {
        'sqs-send': new iam.PolicyDocument({
          statements: [
            new iam.PolicyStatement({
              actions: ['sqs:SendMessage'],
              resources: [sqs.queueArn],
              effect: iam.Effect.ALLOW,
            }),
          ],
        }),
      },
    });
  }
}
```

**Comment by clutteralx** [comment]
Thank you for sharing your implementation @benm5678!
I have been getting this error when trying to add a route to a `HttpApi` using your `HttpSqsSendIntegration`:
<img width="545" alt="firefox_jbkJtzWfeY" src="https://github.com/aws/aws-cdk/assets/71278514/cdc5a31c-5b73-48f0-b076-0ba8cee111c7">

I haven't seen anyone else have this `Mapping expression missing source or destination` error.

**Comment by tomgelmersio** [comment]
@clutteralx This might be because, in my experience, the MessageAttributes cannot be an empty string. Try removing the mapping entirely if there is no value.

**Comment by tomgelmersio** [comment]
Yes thanks for sharing @benm5678, this works great. But I have one problem/question. I can't seem to map headers from the request to the sqs message. When trying the appendheader method shown above i get this error when deploying: "Parameter: append:header.signature does not fit schema for Operation: SQS-SendMessage."
My bind method looks like this:

`public override IHttpRouteIntegrationConfig Bind(IHttpRouteIntegrationBindOptions options)
    {
        var paramMapping = new ParameterMapping()
            .Custom("QueueUrl", _queue.QueueUrl)
            .Custom("MessageBody", _props?.MessageBody ?? "$request.body")
            .AppendHeader("signature", MappingValue.RequestHeader("signature"));

        if (!string.IsNullOrWhiteSpace(_props?.MessageAttributes))
        {
            paramMapping.Custom("MessageAttributes", _props.MessageAttributes);
        }

        return new HttpRouteIntegrationConfig
        {
            Type = HttpIntegrationType.AWS_PROXY,
            Subtype = HttpIntegrationSubtype.SQS_SEND_MESSAGE,
            PayloadFormatVersion = PayloadFormatVersion.VERSION_1_0,
            ParameterMapping = paramMapping,
            Credentials = IntegrationCredentials.FromRole(_role)
        };
    }`

**Comment by clutteralx** [comment]
> @clutteralx This might be because, in my experience, the MessageAttributes cannot be an empty string. Try removing the mapping entirely if there is no value.

That was the issue. It works perfectly now. Thanks so much 🎉
